### DISCUSSION TITLE
What logging framework is recommended for NextJS server & client side logging

### DISCUSSION THREAD

**Discussion by kirankulkarni** [vercel/next.js#58901] [Help]
### Summary

I have built a new app using NextJS 14 and have been hosting it on fly.io I could find several solutions for logging on the server components & server actions. However, none of them help with client component logging.  Please recommend logger that will be able to log data from server components, server actions & client components. I plan to use Vector to accumulate and sink all logs to Graphana loki eventually.

**Comment by ishaan812** [comment]
anything you found?

**Comment by HakimAbdul12** [comment]
check pino out
